documentation/topics/graphql-generation.md

```diff
@@ -168,6 +170,124 @@ query($representative_id: ID) {
 }
 ```
 
+## Mutations and Enums
+
+Now, let's look at how to create a ticket by using a GraphQL mutation.
+
+Let's say you have a Resource that defines an enum-like attribute:
+
+```elixir
+defmodule Helpdesk.Support.Ticket do
+  use Ash.Resource,
+    ...,
+    extensions: [
+      AshGraphql.Resource
+    ]
+
+  alias Helpdesk.Support.TicketStatus
+
+  attributes do
+    uuid_primary_key :id
+    attribute :subject, :string
+    attribute :status, TicketStatus
+  end
+
+  actions do
+    defaults [:create, :read, :update, :destroy]
+  end
+
+  graphql do
+    type :ticket
+
+    queries do
+      get :get_ticket, :read 
+    end
+    
+    mutations do
+      create :create_ticket, :create
+    end
+  end
+end
+```
+
+Above, the following changes have been added:
+
+1. In the `attributes` section, the `:status` attribute has been added. This is an enum whose type
+   is defined in `TicketStatus`. We'll look at that module in a moment.
+2. In the `actions` section, the `:create` action has been added.
+3. The `:create_ticket` mutation has been defined in the `graphql` section.
+
+Here's the `TicketStatus` module:
+
+```elixir
+defmodule Helpdesk.Support.TicketStatus do
+  use Ash.Type.Enum,
+    values: [:open, :closed]
+
+  def graphql_type, do: :ticket_status
+end
+```
+
+`TicketStatus` defines the Ash enum type and the corresponding GraphQL enum type. Without AshGraphql,
+the ticket `:status` attribute could be defined more simply as:
+  
+```elixir
+attribute :status, one_of: [:open, :closed]
```

Comment:
```suggestion
attribute :status, :atom, constraints: [one_of: [:open, :closed]]
```

Reply:
You can technically do it that way, FWIW, and AshGraphql will define an enum from it. I'm pretty sure it would even be called `ticket_status`. But having a separate type for it is a bit cleaner/more reusable.